examples/multisig/methods/guest/src/bin/zk_multisig.rs

```diff
@@ -0,0 +1,345 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_main]
+
+use hex::FromHex;
+use k256::ecdsa::{RecoveryId, Signature, VerifyingKey};
+use risc0_zkvm::guest::env;
+use rs_merkle::{algorithms::Sha256, Hasher, MerkleProof, MerkleTree};
+use sha3::{Digest, Keccak256};
+
+risc0_zkvm::guest::entry!(main);
+
+const TRUSTED_MERKLE_ROOT: &str =
+    "1962fbc17a4440d383f544724e9a943773c0cc6eade89b66b724c31cf33d3f62";
+
+pub fn main() {
+    let signature_one_input: String = env::read(); // <ecdsa_sig_address_one + recovery_id>
+    let signature_two_input: String = env::read(); // <ecdsa_sig_address_two + recovery_id>
+    let merkle_proof_one_input: String = env::read(); // <proof_hashes + index_of_leaf>
+    let merkle_proof_two_input: String = env::read(); // <proof_hashes + index_of_leaf>
+    let signed_message_input: String = env::read(); // <signed_message>
+    let all_leaves_input: String = env::read(); // <merkle_leaves>
```

Comment:
We could use a single struct and have all of these be fields of the struct, that way this could would be `let input: Input = env::read()`